### DISCUSSION TITLE
Some support with how I utilize the API for a Telegram Bot

### DISCUSSION THREAD

**Discussion by schemar** [bafolts/terraforming-mars#3330] [General]
Hey there,

over at https://github.com/schemar/terraforming-mars-turn-bot I have very quickly hacked code that I use for a Telegram Bot to notify my when it's my turn in one of the games.

It isn't 100% reliable to inform users, especially when drafting. Am I using/interpreting the API wrong?

This is what I currently do:

* check `/api/waitingfor?id...`
* when `WAIT`: do nothing
* when `REFRESH`:
    * update game age and undo count from `/api/player?id...`
    * try again
* when `GO`:
    * compare with latest game age when user was informed (also undo count)
    * if lower than current game age:
        * inform user that it is their turn
        * update latest game age when user was informed to current game age (also undo count)
    * else:
        * do nothing

Thank you so much for your time!

**Comment by bafolts** [comment]
During a `GO` you will also have to `update game age and undo count from /api/player?id...`. If the results of `/api/waitingfor?id...` is `GO` that player must take some action. When you call `GO` the result of the call to /api/player should never be less than the game age on the client.

**Reply by schemar** [reply]
Thank you for the quick response! I will try that. 

For context: It’s not a full game client. We play async and the bot simply sends me a telegram message with the link to the game when it is my turn.

**Reply by schemar** [reply]
OK, thank you. I think I understand the API and how it is supposed to be used better, now. My bot is not a game client and therefore it doesn’t know when the user is done taking their turn. I think the quick solution is to always poll `/api/player` instead of `/api/waitingfor`.

Would you accept a PR that adds a new endpoint to the server that would return `GO` or `WAIT` and the current game age? Then a bot could determine whether a `GO` for the current game age was already communicated. Or does the game age also increase within a single player’s turn? What about drafting?

Thank you! If you don’t have the time, just ignore me 😊 I will use the `player` API. I think that’s fine.
